app/src/main/java/com/example/calculator/CommandProcessor.kt

```diff
@@ -0,0 +1,168 @@
+package com.example.calculator
+
+import java.math.BigDecimal
+
+interface Command {
+    fun execute(): BigDecimal?
+}
+
+sealed class CommandResult {
+    object Success : CommandResult()
+    class Error(val message: String): CommandResult()
+
+    companion object {
+        const val WRONG_INPUT = "Wrong input! Please try again."
+        const val DIV_BY_ZERO = "Dividing by zero is forbidden! Please try again."
+    }
+}
+
+class CommandProcessor {
+    private val queue = mutableListOf<Command>()
+    private var result: BigDecimal? = BigDecimal(0)
```

Comment:
```suggestion
    private var result: BigDecimal = BigDecimal.ZERO
```

и чтобы убрать метод getResult
```suggestion
    var result: BigDecimal? = BigDecimal.ZERO
        private set
```